jdk17.sh

```diff
@@ -2,8 +2,21 @@
 set -euxo pipefail
 
 apt-get update
+apt-get -y dist-upgrade
 DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
-  openjdk-17-jdk-headless \
-  gradle
+  curl openjdk-17-jdk-headless unzip
 apt-get clean
 rm -rf /var/lib/apt/lists/*
+
+curl -sLo/tmp/gradle.zip https://services.gradle.org/distributions/gradle-7.4.2-bin.zip
```

Comment:
This is now the only place in this repository where the hash of a downloaded file isn't checked.

Reply:
Thanks for the remark, I'll fix